src/core/reducers/addons.js

```diff
@@ -37,69 +58,151 @@ export function fetchAddon({ errorHandler, slug }: FetchAddonParams): FetchAddon
   };
 }
 
-export function getGuid(result: AddonType) {
-  if (result.type === ADDON_TYPE_THEME) {
+export function getGuid(addon: ExternalAddonType): string {
+  if (addon.type === ADDON_TYPE_THEME) {
     // This mimics how Firefox appends @personas.mozilla.org internally.
     // It's needed to look up themes in mozAddonManager.
-    return `${result.id}@personas.mozilla.org`;
+    return `${addon.id}@personas.mozilla.org`;
   }
-  return result.guid;
+  return addon.guid;
+}
+
+export function removeUndefinedProps(object: Object): Object {
```

Comment:
Is this method needed because there are spreads in different places? Otherwise I would argue that the state should have a consistent shape.

Reply:
Yes it is needed for spreads. I added a comment about it below. It's going to be hard to remove the spreads and I think it's a low priority fix.

Reply:
> I think it's a low priority fix.

👍